qtwebengine/src/core/content_browser_client_qt.cpp

```diff
@@ -217,24 +222,44 @@ class MessagePumpForUIQt : public QObject,
 
 private:
     bool handleScheduledWork() {
-        bool more_work_is_plausible = m_delegate->DoWork();
-
-        base::TimeTicks delayed_work_time;
-        more_work_is_plausible |= m_delegate->DoDelayedWork(&delayed_work_time);
-
-        if (more_work_is_plausible)
-            return true;
-
-        more_work_is_plausible |= m_delegate->DoIdleWork();
-        if (!more_work_is_plausible)
-            ScheduleDelayedWork(delayed_work_time);
-
-        return more_work_is_plausible;
+        unsigned int num_completed(0);
+        bool did_work(false);
+
+        while (num_completed < m_eventCount) {
+            num_completed++;
```

Comment:
*nitpick* Is it possible for multiple events to be completed each loop? I guess it just comes down to variable naming, but it seems like the variable is actually counting the number of times we pump the event loop rather than the number of events that were processed. Maybe I'm just misunderstanding what `DoWork`, `DoDelayedWork` and `DoIdleWork` do.

Reply:
Technically it is possible to complete one work item and one delayed work item per iteration. I renamed `number_completed` to `loop_count`, but I'm having a hard time coming up with a better name for `m_eventCount`.